src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -330,21 +533,45 @@ class InventoryRowModel extends DefaultTableModel {
 
         private static final long serialVersionUID = 1L;
 
+        public InventoryRowModel() {
+            if (allRadio.isSelected()) {
+                try {
+                    inventoryRowList = loadNewInventoryTable(null);
+                } catch (OHServiceException e) {
+                    inventoryRowList = new ArrayList<>();
+                    OHServiceExceptionUtil.showMessages(e);
+                }
+            } else if (specificRadio.isSelected() && code != null && !code.trim().equals("")) {
```

Comment:
If you think the `code` can have trailing spaces then why do you use it in line 546 with those trailing spaces present?

Reply:
Please, I don't quite understand your question

Reply:
You trim the `code` field to test it:   `!code.trim().equals("")`.   This indicates to me that you expect trailing spaces could be present.

Then you use it in line 546 without trimming:  `loadNewInventoryTable(code);`  Here spaces could or could not be present.  One would not expect that the number of trailing spaces would be significant.

Search for `trim()` in this PR and you will see other occasions of trimming the value before using it in a method call.

Reply:
ok